genesis/process/genesisBlockCreator.go

```diff
@@ -29,11 +34,58 @@ func NewGenesisBlockCreator(arg ArgsGenesisBlockCreator) (*genesisBlockCreator,
 		return nil, fmt.Errorf("%w while creating NewGenesisBlockCreator", err)
 	}
 
-	return &genesisBlockCreator{
+	gbc := &genesisBlockCreator{
 		arg:                 arg,
 		shardCreatorHandler: CreateShardGenesisBlock,
 		metaCreatorHandler:  CreateMetaGenesisBlock,
-	}, nil
+	}
+
+	if arg.HardForkConfig.MustImport {
+		err := gbc.createHardForkImportHandler()
```

Comment:
we should think of a way to invalidate the must import flag after the first successful sync of the network. Might use a max nonce until this flag is validated?

Reply:
after a hardfork everybody has to start and create the genesis block through this method - so it is ok. MustImport as a flag will be set only when there is a hardfork - for that you need to set export and import storage / folder.